File: bl-central-server/bl-web-app/src/main/java/org/jesperancinha/logistics/web/data/BridgeOpeningConflictDto.java
package org.jesperancinha.logistics.web.data;

import lombok.Builder;

import java.util.List;

/**
 * This object is made when a conflict is found.
 * It will accumulate all conflicts found in reference to the first element in the list
 */
public record BridgeOpeningConflictDto(String message,
                                       List<BridgeOpeningTimeDto> relatedOpeningTimes) {
    @Builder
    public BridgeOpeningConflictDto {
    }
}

File: bl-central-server/bl-web-app/src/main/kotlin/org/jesperancinha/logistics/web/data/BridgeOpeningConflictDto.kt
package org.jesperancinha.logistics.web.data

import lombok.Builder

/**
 * This object is made when a conflict is found.
 * It will accumulate all conflicts found in reference to the first element in the list
 */
data class BridgeOpeningConflictDto(
    val message: String? = null,
    val relatedOpeningTimes: MutableList<BridgeOpeningTimeDto>
)